src/marqo/tensor_search/web/api_utils.py

```diff
@@ -129,3 +133,48 @@ def decode_mappings(mappings: Optional[str] = None) -> dict:
             return as_dict
         except json.JSONDecodeError as e:
             raise InvalidArgError(f"Error parsing mappings. Message: {e}")
+
+
+def add_docs_params_orchestrator(index_name: str, body: Union[AddDocsBodyParamsOld, AddDocsBodyParamsNew],
+                                device: str, auto_refresh: bool = True, non_tensor_fields: Optional[List[str]] = [],
+                                mappings: Optional[dict] = dict(), model_auth: Optional[ModelAuth] = None,
+                                image_download_headers: Optional[dict] = dict(),
+                                use_existing_tensors: Optional[bool] = False, query_parameters: Optional[Dict] = dict()) -> AddDocsParams:
+    """An orchestrator for the add_documents API to support both old and new versions of the API.
+    All the arguments are decoded and validated in the API function. This function is only responsible for orchestrating.
+
+    Returns:
+        AddDocsParams: An AddDocsParams object for internal use
+    """
+
+    if isinstance(body, AddDocsBodyParamsNew):
+        docs = body.documents
+
+        # Check for query parameters that are not supported in the new API
+        deprecated_fields = ["non_tensor_fields", "use_existing_tensors", "image_download_headers", "model_auth", "mappings"]
+        if any(field in query_parameters for field in deprecated_fields):
+            raise BadRequestError("Marqo is not accepting any of the following parameters in the query string: "
+                                  "`non_tensor_fields`, `use_existing_tensors`, `image_download_headers`, `model_auth`, `mappings`. "
+                                  "Please move these parameters to the request body as "
+                                  "`nonTensorFields, useExistingTensors, imageDownloadHeaders, modelAuth, mappings`. and try again. "
+                                  "Please check `https://docs.marqo.ai/latest/API-Reference/documents/` for the correct APIs.")
+
+        mappings = body.mappings
+        non_tensor_fields = body.nonTensorFields
+        use_existing_tensors = body.useExistingTensors
+        model_auth = body.modelAuth
+        image_download_headers = body.imageDownloadHeaders
+
+    elif isinstance(body, AddDocsBodyParamsOld):
+        docs = body.__root__
+
+    else:
+        raise BadRequestError("Invalid request body type for `/documents` API. "
```

Comment:
This looks like a runtime error where if it happens, there's an issue with our code and not the request. I assume FastAPI would catch a bad request before it reaches here?

Reply:
yes, a 422 unprocessable entity error will be raised. updateding now